code/drasil-theory/Theory/Drasil/ModelKinds.hs

```diff
@@ -1,42 +1,63 @@
+{-# LANGUAGE TemplateHaskell, Rank2Types, ScopedTypeVariables, PostfixOperators  #-}
+module Theory.Drasil.ModelKinds (ModelKinds(..), RealmVariant(..),
+  setMk, elimMk, lensMk, getEqModQds) where
 
-{-# LANGUAGE TemplateHaskell, Rank2Types, ScopedTypeVariables  #-}
-module Theory.Drasil.ModelKinds (ModelKinds(..), setMk, elimMk, lensMk, getEqModQds) where
-
-import Language.Drasil (QDefinition, RelationConcept, ConceptDomain(..),
-                        Definition(..), ExprRelat(..), Idea(..), 
-                        NamedIdea(..), HasUID(..))
-import Control.Lens (Lens', (^.), to, lens, set, makeLenses, Getter, Setter')
+import Control.Lens ((^.), to, lens, set, makeLenses, makeLensesFor,
+  Getter, Lens', Setter')
 import Data.Maybe (mapMaybe)
+import qualified Data.List.NonEmpty as NE
+
+import Language.Drasil (($=), sy, Expr, RelationConcept,
+  NamedIdea(..), HasUID(..), ExprRelat(..), ConceptDomain(..), Definition(..),
+  Idea(..), DefiningExpr(..), UID, Sentence, QDefinition, QuantityDict)
+
+data RealmVariant = RV {
```

Comment:
Can you add some documentation as to what is meant by this? I'm not even sure I know what the `Expr` here is supposed to represent!

Reply:
Yes, I should definitely do that. I'll merge in master and add some documentation for this. I'll let you know when I've done so.

Reply:
I added docs with the proto-Realms as described below, but the meaning of things will still be the same.